syntax = "proto3";

package datasource_rpc;
import "google/protobuf/empty.proto";
import "google/protobuf/any.proto";
import "forestadmin/rpc_common/proto/forest.proto";

service DataSource {
    // Define your RPC methods here
    rpc Schema (google.protobuf.Empty) returns (SchemaResponse);
    rpc RenderChart (DatasourceChartRequest) returns (google.protobuf.Any);
    rpc ExecuteNativeQuery (NativeQueryRequest) returns (google.protobuf.Any);
}

message SchemaResponse{
    DataSourceSchema Schema = 1;
    repeated CollectionSchema Collections = 2;
}

message DatasourceChartRequest{
    Caller caller = 1;
    string ChartName = 2;
}


message NativeQueryRequest{
    Caller Caller = 1;
    string Query = 2;
    map<string, google.protobuf.Any> Parameters = 3;
}